Add UserAuth containers (#75)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/user-auth/containers/sign-up.mdx

---
title: SignUp Container
description: Learn about the SignIn container in the UserAuth dropin.
---

import Diagram from '@components/Diagram.astro';
import Callouts from '@components/Callouts.astro';
import OptionsTable from '@components/OptionsTable.astro';
import Aside from '@components/Aside.astro';
import { Steps } from '@astrojs/starlight/components';

> The SignUp container provides a form where the user enter the details needed to create a new account. The execution of this form can vary, depending on how Commerce is configured:

Reviewer: ```suggestion
The SignUp container provides a form where the user enters the details required to create a new account. The execution of this form can vary, depending on how Commerce is configured:
```

The user is logged on when the `isAutoSignInEnabled` property is passed.

If the `routeRedirectOnSignIn` property is passed, the user is redirected to the specified URL after a successful login. Neither the SuccessNotification container nor slots willv be rendered.

In this scenario, email confirmation must be disabled.

<Diagram caption="SignIn standard login container">
  ![SignUp container](@images/user-auth/sign-up-container.png)
</Diagram>

## Registration with email confirmation enabled

If a user creates an account when email confirmation is enabled, a different form is rendered in place of the sign-up form to inform the user about the next steps. Automatic sign-in is not possible if email confirmation is enabled.

<Diagram caption="SignIn standard login container">
  ![SignUp container](@images/user-auth/sign-up-verify-email.png)
</Diagram>

## SignUp configurations

The SignUp container provides the following configuration options:

<OptionsTable
    compact
    options={[
        ['Options', 'Type', 'Req?', 'Description'],

>         ['addressesData', 'AddressFormProps[]', 'No', 'Allows passing address data into the SignUp drop-in. On successful sign-up, corresponding addresses will be automatically created in the newly created account.'],

Reviewer: ```suggestion
        ['addressesData', 'AddressFormProps[]', 'No', 'Allows passing address data into the SignUp dropin. On successful sign-up, corresponding addresses will be automatically created in the newly created account.'],
```

        ['inputsDefaultValueSet', 'inputsDefaultValueSetProps[]', 'No', ' Allows passing default values for inputs (prefilled sign-up form).'],

>         ['fieldsConfigForApiVersion1', 'any', 'No', 'Allows passing a set of field configurations for environments that do not support the createCustomerV2 mutation (fallback, should not be used in environments with the latest AC version).'],

Reviewer: ```suggestion
        ['fieldsConfigForApiVersion1', 'any', 'No', 'Allows passing a set of field configurations for environments that do not support the createCustomerV2 mutation (fallback, should not be used in environments with the latest Commerce version).'],
```

>         ['apiVersion2', 'boolean', 'No', 'Defaults to true. Allows switching to the createCustomer mutation if createCustomerV2 is not supported in the current version of AC (fallback, should not be used in environments with the latest AC version).'],

Reviewer: ```suggestion
        ['apiVersion2', 'boolean', 'No', 'Defaults to true. Allows switching to the createCustomer mutation if createCustomerV2 is not supported in the current version of Commerce (fallback, should not be used in environments with the latest Commerce version).'],
```

>         ['displayTermsOfUseCheckbox', 'boolean', 'No', 'Controls displaying of the appropriate checkboxes in the sign-up form. Note: Rendered checkboxes are not functional and should not be used yet. They will be used in the future.'],

Reviewer: ```suggestion
        ['displayTermsOfUseCheckbox', 'boolean', 'No', 'Controls displaying the appropriate checkboxes in the sign-up form. Rendered checkboxes are not functional and should not be used yet. They will be used in the future.'],
```

>         ['displayNewsletterCheckbox', 'boolean', 'No', ''],

Reviewer: The description here is blank.

>         ['isAutoSignInEnabled', 'boolean', 'No', 'Determines if the user should be automatically signed in after registration. This has no effect if email confirmation is enabled in the admin panel (automatic sign-in is not possible in this case).'],

Reviewer: ```suggestion
        ['isAutoSignInEnabled', 'boolean', 'No', 'Determines if the user should be automatically signed in after registration. This has no effect if email confirmation is enabled in the Commerce Admin (automatic sign-in is not possible in this case).'],
```

        ['formSize', 'default | small', 'No', 'Controls form paddings and spacing. Use "small" to embed the form in small layout containers like a dropdown in the header.'],
        ['hideCloseBtnOnEmailConfirmation', 'boolean', 'No', 'Controls the visibility of the “Close” button on the email confirmation view.'],
        ['routeRedirectOnEmailConfirmationClose', 'function', 'No', 'Determines where the user is redirected when the “Close” button on the email confirmation view is clicked.'],
        ['slots', 'function', 'No', 'Allows passing the SuccessNotification container or custom component rendered on successful sign-up if routeRedirectOnSignIn is not provided.'],
        ['routeSignIn', 'function', 'No', 'Determines where the “Already a member? Sign in” link redirects the customer.'],

>         ['routeRedirectOnSignIn', 'function', 'No', 'Determines the page to which the user should be redirected after sign-in (if isAutoSignInEnabled is true). If provided, the user won’t see SuccessNotification or any override provided via slot.'],

Reviewer: ```suggestion
        ['routeRedirectOnSignIn', 'function', 'No', 'Determines the page to which the user should be redirected after sign-in (if isAutoSignInEnabled is true). If provided, the user won’t see SuccessNotification or any override provided through slot.'],
```

        ['onErrorCallback', 'function', 'No', ' Callback executed when an error occurs, receiving the error object as a parameter.'],

>         ['onSuccessCallback', 'function', 'No', 'Callback executed when the user successfully signs up, receiving userName and status as parameters.'],

Reviewer 2: Additionally, we have introduced a new prop responsible for the "Confirm Password" field.

`requireRetypePassword`

PR with screenshot - https://git.corp.adobe.com/Commerce-Storefront/storefront-auth/pull/58
